resources/js/src/app/components/common/LazyMountIntersect.vue

```diff
@@ -0,0 +1,76 @@
+<script>
+    export default {
+        props:
+        {
+            threshold: {
+                type: Number,
+                default: 0.1
+            }
+        },
+
+        data()
+        {
+            return {
+                isVisible: false,
+                loadingRendered: false
+            }
+        },
+
+        created()
+        {
+            this.featureEnabled = 'IntersectionObserver' in window;
```

Comment:
`featureEnabled` ist nicht in `data` definiert.

Reply:
`featureEnabled` ist extra nicht in `data` definiert. Diese Variable muss nicht reactive sein, deshalb wurde sie dort nicht hinterlegt.
Ich kann verstehen, wenn man sie für die DevTools in data haben will. In dem Fall würde ich sie dort hinzufügen.

Reply:
Es geht nicht um unbedingt um die DevTools, vielmehr geht es mir um die Nachvollziehbarkeit. Die Variable wird halt nirgends "deklariert", sodass es beim Lesen des Codes schwer nachzuvollziehen ist, wo die Variable herkommt bzw. wo sie verwendet werden kann. Was hältst du alternativ davon, das Flag als Konstante außerhalb der Komponente zu halten?

```
const FEATURE_ENABLED = 'IntersectionObserver' in window;

export default {
  created()
  {
    if (FEATURE_ENABLED) { ... }
  }
};
```

Reply:
Das klingt gut, wo würden wir diese Flag bevorzugt vorhalten wollen?